viewing_party/party.py

```diff
@@ -1,23 +1,207 @@
+        if movies["title"] == title:
+            user_data["watched"].append(movies)
+            movies_to_remove.append(movies)
+    
+    for movie in movies_to_remove:
+        user_data["watchlist"].remove(movies)
+
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_date):
+    '''
+    receives user_data dict and returns average of all movies ratings in watched list
+    '''
+    counter = 0
+    sum = 0
+    for movie in user_date["watched"]:
+        sum += movie["rating"]
+        counter += 1
+    
+    if counter != 0:
+        return sum/counter
+    else:
+        return 0
+
+# from collections import Counter
+
+def get_most_watched_genre(user_data):
+
+    # if not user_data["watched"]:
+    #     return None
+    # genre_counter_list = [movie["genre"] for movie in user_data["watched"]]
+    # genre_counter = Counter(genre_counter_list)
+    # most_watched_genre = genre_counter.most_common(1)[0][0]
+    # return most_watched_genre
+
+
+    genre_frequency = {}
+
+    for movie in user_data["watched"]:
+        if movie["genre"] not in genre_frequency:
+            genre_frequency[movie["genre"]] = 1
+        else:
+            genre_frequency[movie["genre"]] += 1
+    
+    most_watched_genre = None
+    most_watched_freq = 0
+
+    for genre, frequency in genre_frequency.items():
+        if frequency > most_watched_freq:
+            most_watched_freq = frequency
+            most_watched_genre = genre
+    
+    return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    user_watched = set()
+    friends_watched = set()
+    for movies in user_data["watched"]:
+        user_watched.add(movies["title"])
+    for each_firend in user_data["friends"]:
+        for friend_movies in each_firend["watched"]:
+            friends_watched.add(friend_movies["title"])
+    movie_title = user_watched - friends_watched
+    
+    unique_movie =[]
+    for movies in user_data["watched"]:
+        if movies["title"] in movie_title :
+            unique_movie.append(movies)
+    
+    return unique_movie
+
+
+
+def get_friends_unique_watched(user_data):
+    user_watched = set()
+    friends_watched = set()
+    for movies in user_data["watched"]:
+        user_watched.add(movies["title"])
+    for each_firend in user_data["friends"]:
+        for friend_movies in each_firend["watched"]:
+            friends_watched.add(friend_movies["title"])
+    movie_title = friends_watched - user_watched
+
+    friends_unique_movie = []
+    # seen_titles = set()
+    for firend in user_data["friends"]:
+        for friend_movies in firend["watched"]:
+            if friend_movies["title"] in movie_title and friend_movies not in friends_unique_movie :

```

Comment:
This compound conditional (and most compound conditionals to be honest) breaks the PEP8 rules of a max of 79 characters per line. When we have a compound conditional like this, we can add in a line break pretty easily with the `\` character. This might look something like: 

```suggestion
            if friend_movies["title"] in movie_title \
                 and friend_movies not in friends_unique_movie:

```